pkg/serviceaccount/legacy.go

```diff
@@ -126,6 +140,30 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 			klog.V(4).Infof("Service account UID no longer matches %s/%s: %q != %q", namespace, serviceAccountName, string(serviceAccount.UID), serviceAccountUID)
 			return nil, fmt.Errorf("ServiceAccount UID (%s) does not match claim (%s)", serviceAccount.UID, serviceAccountUID)
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenTracking) {
+			for _, ref := range serviceAccount.Secrets {
+				if ref.UID == secret.UID {
```

Comment:
uid-matching doesn't work for the references the tokens controller generated, only the name mattered:

https://github.com/kubernetes/kubernetes/blob/d4795e4beced59cf4f698de487b738df18e22adf/pkg/controller/serviceaccount/tokens_controller.go#L457

Update the `TestLegacyServiceAccountTokenTracking` integration test to run with LegacyServiceAccountTokenNoAutoGeneration=false and LegacyServiceAccountTokenTracking=true, and add test code to exercise auto-generated warnings:
* wait for an auto-generated secret
* make a client that authenticates with that token (and captures returned warnings)
* make a request with that token, and ensure a warning is returned

Reply:
done